site/app/models/GradingOrder.php

```diff
@@ -496,18 +496,30 @@ public function getSortedGradedGradeables() {
         );
         $gg_idx = [];
         $unsorted = [];
+        $items = [];
         foreach ($iter as $gg) {
-            $idx = $this->getSubmitterIndex($gg->getSubmitter());
-            //Should never happen, but better to be safe
-            if ($idx === false) {
-                $unsorted[] = $gg;
-            }
-            else {
-                $gg_idx[$idx] = $gg;
+            $items[$gg->getSubmitter()->getId()] = $gg;
+        }
+        $count = 0;
+
+        //Iterate through all sections and their submitters to find this one
+        foreach ($this->section_submitters as $name => $section) {
+            for ($i = 0; $i < count($section); $i++) {
+                $testSub = $section[$i]->getId();
+
+                //Found them
+                if (isset($items[$testSub])) {
+                    $gg_idx[$count] = $items[$testSub];
+                    unset($items[$testSub]);
+                }
+
+                $count++;
             }
         }
-        //Since the array's elements were not added in the same order as the indices, sort to fix it
-        ksort($gg_idx);
+        foreach ($items as $item => $val) {
+            $unsorted[] = $val;
+        }
+
         return array_merge($gg_idx, $unsorted);
```

Comment:
Can use [`array_values`](https://www.php.net/manual/en/function.array-values.php) to avoid the foreach. I'd also suggest doing this all inline to avoid any extra memory usage if these are big arrays.

```suggestion
        return array_merge($gg_idx, array_values($items));
```